src/tests/Microsoft.PowerFx.Interpreter.Tests/DeferredTypeTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    public class DeferredTypeTests
+    {
+        [Theory]
+        [InlineData("X")]
+        [InlineData("$\"Test {X}\"")]
+        [InlineData("!X")]
+        [InlineData("Table(X)")]
+        [InlineData("X.Field1")]
+        [InlineData("{test: X}")]
+        [InlineData("RX.XF")]
+        [InlineData("RX.XF + 1")]
+
+        [InlineData("X + 1")]
+        [InlineData("X + \"1\"")]
+        [InlineData("X + DateTime(2022, 11, 10, 0, 0, 0)")]
+        [InlineData("X + Date(2022, 11, 10)")]
+        [InlineData("X + Time(0, 0, 0)")]
+        [InlineData("1 + X")]
+        [InlineData("DateTime(2022, 11, 10, 0, 0, 0) + X")]
+        [InlineData("Date(2022, 11, 10) + X")]
+        [InlineData("Time(0, 0, 0) + X")]
+
+        [InlineData("X * 1")]
+        [InlineData("And(X, 1=1)")]
+        [InlineData("X&\"test\"")]
+        [InlineData("X = 1")]
+        [InlineData("X = GUID(\"5cc45615-f759-4a53-b225-d3a2497f60ad\")")]
+
+        [InlineData("X < 2")]
+        [InlineData("X < DateTime(2022,11,10,0,0,0)")]
+        [InlineData("\"The\" in X")]
+        [InlineData("123 in X")] // check
+        [InlineData("X in [1, 2, 3]")] // check
+        [InlineData("123 in Table(X)")]
+        [InlineData("IsBlank(X)")]
+        [InlineData("IsError(X)")]
+        [InlineData("Text(X)")]
+        [InlineData("Value(X)")]
+        [InlineData("Boolean(X)")]
+        [InlineData("Index([1,2,3].Value, X)")]
+
+        // Ensures expression binds without any errors - but issues a warning for the deferred(unknown) type.
+        public void DeferredTypeTest(string script)
```

Comment:
we should add some tests with overload resolution. 
Sum, Filter 

Sum(?,?) < 5 // succeed, with warning 
First(Sum(?,?)).Field // also succeed, with waring